### DISCUSSION TITLE
best way to apply the same agent to multiple tables?

### DISCUSSION THREAD

**Discussion by Aariq** [rstudio/pointblank#367] [Q&A] [answered]
If I have multiple data frames that all should have the same structure, but I don't necessarily want to combine them, is there a way to create an agent that is table agnostic and apply it to multiple tables?  My current approach is to use `imap()` from the `purrr` package, but it doesn't seem to work with knitted RMarkdown documents.  

```
library(pointblank)
library(tidyverse)

small_table %>%
  group_by(f) %>% 
  group_split() %>%
  set_names(c("high", "low", "mid")) %>% 
  imap(~{
    .x %>%
      create_agent(tbl_name = .y) %>% 
      col_exists(vars(date, date_time)) %>%
      col_vals_regex(
        vars(b),
        regex = "[0-9]-[a-z]{3}-[0-9]{3}"
      )  %>%
      interrogate()
  })
```

**Comment by rich-iannone** [comment, marked as answer]
Maybe for this situation, the `set_tbl()` function might be useful? Check out https://rich-iannone.github.io/pointblank/reference/set_tbl.html for some examples.

**Reply by Aariq** [reply]
This is what I was looking for.  My one wish is that you could also manually change the `tbl_name` with `set_tbl()`

**Reply by rich-iannone** [reply]
That's a really good idea and I didn't think of it! Would you be okay filing an issue for that? Otherwise I can do that.

**Reply by Aariq** [reply]
Sure!
